Sources/SecureXPC/Server/XPCAnonymousServer.swift

```diff
@@ -0,0 +1,44 @@
+//
+//  XPCAnonymousServer.swift
+//  
+//
+//  Created by Alexander Momchilov on 2021-11-28.
+//
+
+import Foundation
+
+internal class XPCAnonymousServer: XPCServer {
+    private let anonymousListenerConnection: xpc_connection_t
+
+    internal override init() {
+        self.anonymousListenerConnection = xpc_connection_create(nil, nil)
+        super.init()
+
+        // Start listener for the new anonymous connection, all received events should be for incoming client connections
+         xpc_connection_set_event_handler(anonymousListenerConnection, { newClientConnection in
+             // Listen for events (messages or errors) coming from this connection
+             xpc_connection_set_event_handler(newClientConnection, { event in
+                 self.handleEvent(connection: newClientConnection, event: event)
+             })
+             xpc_connection_resume(newClientConnection)
+         })
+         xpc_connection_resume(anonymousListenerConnection)
+    }
+
+    public override func start() -> Never {
+        fatalError("Anonymous services don't need to be `start()`ed.")
+    }
+
+    internal override func acceptMessage(connection: xpc_connection_t, message: xpc_object_t) -> Bool {
+        // Anonymous service connections should only ever passed among trusted parties.
```

Comment:
This feels like a security vulnerability waiting to happen. Why not enforce the same sort of requirements that `XPCMachServer` uses? If someone really doesn't want to make the connection secure they can always pass in a requirement that basically anything will satisfy (like `anchor apple generic`).

Reply:
I don't have a good idea for how to design that API yet. Let's ship as-is, and add this on later.

Reply:
> I don't have a good idea for how to design that API yet.

One approach would be to refactor a portion of `XPCMachServer` into something like `XPCSecureServer` and then have `XPCMachServer` and `XPCAnonymousServer` inherit from it. Alternatively, the code could be factored out into a helper function or two. Comes down to whether we want to embrace composition or inheritance. In general I feel composition is a better fit for Swift, but we're already making use of inheritance to represent servers.

> Let's ship as-is, and add this on later.

Sorry, but this isn't an area I'm willing to compromise on as is security is the central tenet of this framework. I very much appreciate the added functionality you've brought to this framework, but for me to be comfortable adding new code it always needs to adhere to this.

Reply:
I suggest shipping it as-is not because I don't think it's important, but because "out of trunk" development is incredibly costly (in time/effort), and realllllly slows everything down.

How about we ship it as `internal`, and don't expose publically until it has a security-requirements API built in?

> One approach would be to refactor a portion of `XPCMachServer` into something like `XPCSecureServer` and then have `XPCMachServer` and `XPCAnonymousServer` inherit from it.

This is what I had in mind, as well.

It's a massive change though, it would be inappropriate to lump it into a single PR like this

Reply:
Hmmm perhaps a compositional approach of having a `MessageAuthenticator` protocol might be preferable. I'll play around with various approaches.

Reply:
> How about we ship it as internal, and don't expose publically until it has a security-requirements API built in?

That works for me.

> Hmmm perhaps a compositional approach of having a `MessageAuthenticator` protocol might be preferable. I'll play around with various approaches.

Not sure there's much of a benefit to having a protocol for this if it's all purely internal to the framework? But happy to take a look whatever you come up with.

Reply:
Tracking this ins #36

I've updated the visibility, so external API users won't be able to create anonymous services.